homeassistant/components/spaceapi/__init__.py

```diff
@@ -254,7 +255,17 @@ def get_sensor_data(hass, spaceapi, sensor):
         """Get data from a sensor."""
         if not (sensor_state := hass.states.get(sensor)):
             return None
-        sensor_data = {ATTR_NAME: sensor_state.name, ATTR_VALUE: sensor_state.state}
+
+        # SpaceAPI sensor values must be numbers
+        try:
+            state = float(sensor_state.state)
+        except ValueError:
+            state = math.nan
```

Comment:
If we get an unexpected state, it should become "Unkown" in Home Assistant (which is equal to setting the state `None`).

Reply:
Please elaborate, I don't understand at all what you mean here.

Reply:
You set the state to `math.nan`: Why not just `None`?

Reply:
Because `null` is not a valid value for SpaceAPI sensor fields: https://github.com/SpaceApi/schema/blob/359b3a204800b0dd1efcbde51e52e240aed2f6a9/13.json#L289